src/main/java/org/concord/sensor/server/SensorStateManager.java

```diff
@@ -326,11 +332,16 @@ public void doAction(Event message, Entity entity, Transition transition, int ac
 					@Override
 					public void run() {
 						try {
-							readSingleValue();
+							if ((device != null) && device.supportsChannelPolling()) {
+								pollChannelValues();
+							}
+							else {
+								readSingleValue();
+							}
 						} catch (Exception e) {
 							errorCount++;
 							logger.error("Failed to read data from the device!", e);
-							if (errorCount > 5) {
+							if (errorCount > 2) {
```

Comment:
Did you check the history of this to see if we tweaked it at all before? 
It seems like one of those things we might have tweaked after experienced errors.
there might also be references to this errorCount in the sensor-projects that might provide more insight into the magic number 5.

Reply:
I did a search for errorCount on github in our concord-consoritum org and didn't find anything. So that isn't a clue about this magic number 5.

Reply:
Looks like it is just magic. It seems to have been added here: https://github.com/concord-consortium/sensor-connector/commit/f146874451147763f730d297db0485c37545aa87

I do remember some kind of retry counts when working on this years ago. So I'll look a bit more for that.

Reply:
OK searching for retry didn't turn up anything. So this seems like it will be guess and check. Can you add a comment about your rationale for changing it from 5 to 2?